src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/CartDiscountServiceImplIT.java

```diff
@@ -0,0 +1,413 @@
+            .value(CART_DISCOUNT_VALUE_DRAFT_2)
+            .target(CART_DISCOUNT_TARGET_2)
+            .sortOrder(SORT_ORDER_2)
+            .requiresDiscountCode(false)
+            .key(CART_DISCOUNT_KEY_2)
+            .isActive(false)
+            .build();
+
+    final ProjectApiRoot spyClient = spy(CTP_TARGET_CLIENT);
+
+    final CartDiscountSyncOptions spyOptions =
+        CartDiscountSyncOptionsBuilder.of(spyClient)
+            .errorCallback(
+                (exception, oldResource, newResource, actions) -> {
+                  errorCallBackMessages.add(exception.getMessage());
+                  errorCallBackExceptions.add(exception);
+                })
+            .build();
+
+    final CartDiscountService spyCartDiscountService = new CartDiscountServiceImpl(spyOptions);
+
+    // test
+    final Optional<CartDiscount> createdCartDiscount =
+        spyCartDiscountService
+            .createCartDiscount(newCartDiscountDraft)
+            .toCompletableFuture()
+            .join();
+
+    final CartDiscount cartDiscount =
+        CTP_TARGET_CLIENT
+            .cartDiscounts()
+            .withKey(CART_DISCOUNT_KEY_2)
+            .get()
+            .execute()
+            .thenApply(ApiHttpResponse::getBody)
+            .toCompletableFuture()
+            .join();
+
+    assertThat(createdCartDiscount)
+        .hasValueSatisfying(
+            created -> {
+              assertThat(created.getKey()).isEqualTo(cartDiscount.getKey());
+              assertThat(created.getName()).isEqualTo(cartDiscount.getName());
+              assertThat(created.getCartPredicate()).isEqualTo(cartDiscount.getCartPredicate());
+              assertThat(created.getValue()).isEqualTo(cartDiscount.getValue());
+              assertThat(created.getTarget()).isEqualTo(cartDiscount.getTarget());
+              assertThat(created.getSortOrder()).isEqualTo(cartDiscount.getSortOrder());
+            });
+  }
+
+  @Test
+  void createCartDiscount_WithDuplicateCartDiscountKey_ShouldHaveEmptyOptionalAsAResult() {
+    // preparation
+    final CartDiscountDraft newCartDiscountDraft =
+        CartDiscountDraftBuilder.of()
+            .name(CART_DISCOUNT_NAME_2)
+            .cartPredicate(CART_DISCOUNT_CART_PREDICATE_2)
+            .value(CART_DISCOUNT_VALUE_DRAFT_2)
+            .target(CART_DISCOUNT_TARGET_2)
+            .sortOrder(SORT_ORDER_2)
+            .requiresDiscountCode(false)
+            .key(CART_DISCOUNT_KEY_1)
+            .isActive(false)
+            .build();
+
+    final CartDiscountSyncOptions options =
+        CartDiscountSyncOptionsBuilder.of(CTP_TARGET_CLIENT)
+            .errorCallback(
+                (exception, oldResource, newResource, actions) -> {
+                  errorCallBackMessages.add(exception.getMessage());
+                  errorCallBackExceptions.add(exception);
+                })
+            .build();
+
+    cartDiscountService = new CartDiscountServiceImpl(options);
```

Comment:
why creating a new `CartDiscountService` instead of using existing one?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1027/commits/c4b802dbfa39528e0314fe18b21af832e8b0e88c